ngraph/cmake/external_onnx.cmake

```diff
@@ -46,23 +47,40 @@ macro(onnx_set_target_properties)
         target_compile_options(onnx PRIVATE -Wno-unused-variable -Wno-unused-parameter)
         target_compile_options(onnx_proto PRIVATE -Wno-unused-variable)
     endif()
+
+    set_target_properties(onnx_proto PROPERTIES
+        CXX_VISIBILITY_PRESET default
```

Comment:
Can we use visibility hidden?

Reply:
Strange, when I tested it last time linker doesn't see some symbols without such property, but now it works (at least on my dev environment).
done

Reply:
as far as I can see, when we build `onnx_proto`, cmake from onnx defines `ONNX_API` as:
```
DONNX_API="__attribute__((__visibility__(\"default\")))"
```
So, setting visibility almost has no effect since visibility is explicitly set per-symbols (see https://github.com/onnx/onnx/blob/master/CMakeLists.txt#L386)

Reply:
^^good point

Reply:
@ilyachur @ilya-lavrenov 
I am not sure why but there is problem for macos (release and debug) if we don't setting this property.
In CI tests I got the exception (you can look to CI results connected with commit b93e77e):
```
[ RUN      ] ExtensionTests.testGetImplTypes
[libprotobuf ERROR _deps/ext_protobuf-src/src/google/protobuf/descriptor_database.cc:58] File already exists in database: onnx/onnx_ngraph_onnx-ml.proto
[libprotobuf FATAL _deps/ext_protobuf-src/src/google/protobuf/descriptor.cc:1370] CHECK failed: GeneratedDatabase()->Add(encoded_file_descriptor, size): 
libc++abi.dylib: terminating with uncaught exception of type google::protobuf::FatalException: CHECK failed: GeneratedDatabase()->Add(encoded_file_descriptor, size):
```
It is exactly the same as for situation when we set `onnx_proto` to be built as static lib.
What's really strange the exception is thrown during `ieUnitTests` execution which has no dependency to any onnx-related target.
I was not able to reproduce it on our macos dev environment.

Mechanism of register to protobuf database is based on trick that static are initialized as fist. In `onnx_ngraph_onnx-ml.pb.cc` generated file you can find:
```
// Force running AddDescriptors() at dynamic initialization time.
static bool dynamic_init_dummy_onnx_2fonnx_5fngraph_5fonnx_2dml_2eproto = []() { AddDescriptors_onnx_2fonnx_5fngraph_5fonnx_2dml_2eproto(); return true; }();
```

Do you have maybe idea how to deal with it or it is ok to leave this `set_target_properties`?

@tomdol Can you look at it too?

Reply:
I've tried figuring it out but it has to be something mac-specific. I don't know much about this system and apple-clang so it's hard to even suggest anything. 

I had problems with mac in my recent PR too, it was required to keep the linking between the unit tests and protobuf just because mac could find some include dirs. It was working on all other CI configurations though.

Reply:
I think we can keep visibility default for Apple only..

Reply:
for me it is ok, done